verticapy/performance/vertica/collection/profile_import.py

```diff
@@ -54,33 +59,97 @@ def __init__(
 
+        # initialize internal attributes
+        self.tarfile_obj = None
+        self.bundle_version = None
+        self.raise_when_missing_files = False
+        self.skip_create_table = False
+        self.tmp_path = os.getcwd()
+
+    @property
+    def skip_create_table(self) -> bool:
+        """
+        ProfileImport will SKIP running ddl statements to create new
+        tables when skip_creat_table is True. Otherwise it will run
+        DDL statements. Default value is False.
+
+        The DDL statements are:
+           CREATE TABLE ... IF NOT EXISTS
+           CREATE PROJECTION ... IF NOT EXISTS
+        """
+        return self._skip_create_table
+
+    @skip_create_table.setter
+    def skip_create_table(self, val: bool) -> None:
+        if not isinstance(val, bool):
+            raise ValueError(
+                f"Cannot set skip_create_table to value of type {type(val)}. Must be type bool."
+            )
+        self._skip_create_table = val
+
+    @property
+    def raise_when_missing_files(self) -> bool:
+        """
+        ProfileImport will raise and exception if a bundle lacks
+        any of the expected files and raise_when_missing_files is True.
+        Otherwise, it will write a warning to the log. Default value is False.
+        """
+        return self._raise_when_missing_files
+
+    @raise_when_missing_files.setter
+    def raise_when_missing_files(self, val: bool) -> None:
+        if not isinstance(val, bool):
+            raise ValueError(
+                f"Cannot set raise_when_missing_files to value of type {type(val)}. Must be type bool."
+            )
+        self._raise_when_missing_files = val
+
+    @property
+    def tmp_path(self) -> Path:
+        """
+        ProfileImport will raise and exception if a bundle lacks
+        any of the expected files and raise_when_missing_files is True.
+        Otherwise, it will write a warning to the log. Default value is False.
+        """
+        return self._tmp_path
+
+    @tmp_path.setter
+    def tmp_path(self, val: os.PathLike) -> None:
+        if not isinstance(val, str) and not isinstance(val, os.PathLike):
+            raise ValueError(
+                f"Cannot set tmp_dir to value of type {type(val)}. Must be type string or PathLike."
+            )
+        self._tmp_path = val
+
     def check_file(self) -> None:
         """
         Checks to see that the file exists and that we can open it
         """
         if not os.path.exists(self.filename):
             raise FileNotFoundError(f"File {self.filename} does not exist")
 
-        # Recall that open will raise errors if
-        # the file does not exist
-        self.input_file_obj = open(self.filename, "r")
+        unpack_dir = self._unpack_bundle()
+        self.bundle_version = self._calculate_bundle_version(unpack_dir)
+        self._check_for_missing_files(unpack_dir, self.bundle_version)
+        self._load_vdataframes(unpack_dir, self.bundle_version)
```

Comment:
Doesn't make sense to load all VDFs as part of  check_file? It sounds like a time consuming operation more than a check.

Reply:
Refactored `_load_vdataframes` so that it is done as part of a new function, `load_file`